### ISSUE TITLE
Software architecture design

### ISSUE THREAD

**Issue by qy21gafy** [amosproj/amos2022ws05-shared-desk-mgmt#30]
## User story
1. As a software developer
2. I want / need a base architecture to exist for the project
3. So that I can add 'actual' features for the users

## Acceptance criteria
* Architecture should consider frontend, backend and communication
* Architecture should consider user and admin roles existing in the final product
* Architecture should allow future changes and the integration of new components

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
